File: Linguaggio_Alto_Livello/Java/Hard/PrimeSieve.java
import java.util.ArrayList;
import java.util.List;

public class PrimeSieve {
    public static void main(String[] args) {
        int limit = 50;
        boolean[] isPrime = new boolean[limit + 1];
        for (int i = 0; i <= limit; i++) {
            isPrime[i] = true;
        }
        isPrime[0] = false;
        isPrime[1] = false;

        for (int number = 2; number * number <= limit; number++) {
            if (isPrime[number]) {
                for (int multiple = number * number; multiple <= limit; multiple += number) {
                    isPrime[multiple] = false;
                }
            }
        }

        List<Integer> primes = new ArrayList<>();
        for (int i = 2; i <= limit; i++) {
            if (isPrime[i]) {
                primes.add(i);
            }
        }

        System.out.println("Numeri primi fino a 50:");
        System.out.println(primes);
    }
}

File: Linguaggio_Medio_Livello/C/Hard/prime_sieve.c
#include <stdio.h>
#include <stdbool.h>

int main(void) {
    int limit = 50;
    bool is_prime[51];

    for (int i = 0; i <= limit; i++) {
        is_prime[i] = true;
    }
    is_prime[0] = false;
    is_prime[1] = false;

    for (int number = 2; number * number <= limit; number++) {
        if (is_prime[number]) {
            for (int multiple = number * number; multiple <= limit; multiple += number) {
                is_prime[multiple] = false;
            }
        }
    }

    printf("Numeri primi fino a 50:\n");
    for (int i = 2; i <= limit; i++) {
        if (is_prime[i]) {
            printf("%d ", i);
        }
    }
    printf("\n");

    return 0;
}
